src/rt/aaA.d

```diff
@@ -533,12 +533,16 @@ extern (C) void* _aaGetY(AA* aa, const TypeInfo_AssociativeArray ti,
  *      If key was not in the aa, a mutable pointer to newly inserted value which
  *      is set to all zeros
  */
-extern (C) void* _aaGetX(AA* aa, const TypeInfo_AssociativeArray ti,
+extern (C) void* _aaGetX(AA* paa, const TypeInfo_AssociativeArray ti,
     const size_t valsz, scope const void* pkey, out bool found)
 {
     // lazily alloc implementation
-    if (aa.impl is null)
-        aa.impl = new Impl(ti);
+    AA aa = *paa;
+    if (aa is null)
```

Comment:
Why this?

Reply:
What do you mean? If `AA == Impl*`, there's of course no `.impl`.

Reply:
You mean you're just taking advantage of the `alias this`?

Reply:
Yes. The `paa` to `aa` deref is required, because previous `paa.firstUsed` was a double-deref ('paa->impl->firstUsed').